### DISCUSSION TITLE
"The job has failed" in running example in DOSXYZnrc

### DISCUSSION THREAD

**Discussion by fatihahjazaaufa** [nrc-cnrc/EGSnrc#1075] [Questions] [answered]
I have installed EGSnrc and want to try the DOSXYZnrc. I try to run the examples. Every example I used, it compiled successfully. But when I run it, it always said "The job has failed". There are no further message on what has been wrong. I already check the installation process and everything seems right. Is there anything that I could do? Because it also happen in my friend's laptop

The OS I used is windows. Here is the process

![PIC 1](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/f81ade23-f81c-4fdc-99d1-5e0af28c10bd)
![PIC 2](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/ecfa0800-89f8-4a63-a474-6999e9a73418)
![PIC 3](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/662679c5-9dc0-4b04-b5d9-c489afcb91e5)
![PIC 4](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/1122bc2a-694b-44c6-987f-1a5f084e745c)
![PIC 5](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/9463ceea-5931-415d-a1c8-4d30099be7ea)

and this happen when I run it
![PIC 6](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/9e3f23d5-81db-483a-a408-9e8fef6d1ab5)

Thank you

**Comment by crcrewso** [comment]
First thing I noticed, I believe AIR700ICRU requires the 700ICRU pegs4dat file. Could you try running it again?

**Comment by rtownson** [comment]
The beamsource dosxyz example requires you to compile the BEAMnrc accelerator that it is using for a source as a shared library. Navigate to `egs_home/BEAM_EX16MVp` on the command line and type `make library` (on linux) or `mingw32-make library` (on windows). Also, compile that BEAMnrc accelerator normally (just `make`), and test a short simulation with the beamnrc input file (`EX16MVp.egsinp`) to ensure that simulation runs properly.

**Comment by fatihahjazaaufa** [comment]
Thank you for the responses! 

I have tried your suggestions @rtownson and here are the results

1. Type mingw32-make library
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/b6f06f55-ff2f-4b8d-8483-1f53ee5c770b)

2. Compile the beam accelerator normally (type make)
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/82c40f1c-ad82-4b7f-b22a-17395d78e316)

Is there anything wrong? also when I test the short simulation using EX16MVp.egsinp it is said that the module file is corrupt. Here are the details

![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/0c5d262f-428c-4ceb-93b8-75a4c92df8a4)

![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/5a2b04e1-fd29-4b6d-aa16-bd9c964cc27f)

![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/f496a611-29e9-43b5-b1b6-24d326948b8f)

Thank you

**Comment by rtownson** [comment]
Step 2: Since you're using mingw32 compilers, you'll need to use `mingw32-make`, not just `make`.

Step 3: Load the .module file, not the .egsinp file when you `Load a previous accelerator`. You select the input file when you use `Load a previous input file`.

**Comment by fatihahjazaaufa** [comment]
I already do the simulation with the EX16MVp and also get the results (output files: .egslst, .egslog, .egsplot, .egsphsp, .egsgeom, .egsgph, .errors). Unfortunately, I still encounter the same problem when try to run example in dosxyznrc (the steps are exactly the same as the fist comment I sent)

Thank you

**Comment by rtownson** [comment]
Ok, please post the dosxyz output, does it print an error? Check the .egslst file. Also, make sure you are using the 700icru pegs file as suggested, not 521icru.

**Comment by fatihahjazaaufa** [comment]
Ok, I already tried it, even the output file (.egslst file) is nowhere to be found. Here are the details
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/4eb8d6a8-52b5-4cb8-99bb-bb8ba234016b)
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/35e145c2-3d49-4ae1-93ae-58ffd02ac74f)
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/a5eb3f63-a4d5-44ff-8c93-22bb0593c507)
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/bce5f079-446c-446d-8e96-d539bbeec292)
Thank you

**Comment by fatihahjazaaufa** [comment]
I also tried to run DOSXYZnrc from the command line, but I still encounter errors
![image](https://github.com/nrc-cnrc/EGSnrc/assets/143057184/cc99dde1-c01e-4a5d-9641-6ca3a88376aa)

thank you

**Comment by rtownson** [comment, marked as answer]
All EGSnrc applications require the input file to be directly in the `egs_home/application/` directory, not in a sub directory. I see the problem is that the example file is in the `examples` directory - you need to copy it up to the `dosxyznrc` directory. Also, when you run on the command line, do not provide file paths. Since it already knows exactly which folder to look in, you just need to type:

```
dosxyznrc -i 16MVp_h2o_phantom_beamsource_example -p 700icru
```

**Comment by fatihahjazaaufa** [comment]
Thank you @rtownson for the help, it works now!
